src/tmt_web/api.py

```diff
@@ -0,0 +1,163 @@
+            str | None,
+            Query(
+                alias="plan-path",
+                title="Plan path",
+                description="Path to the plan metadata directory",
+            ),
+        ] = None,
+        out_format: Annotated[Literal["html", "json", "yaml"], Query(alias="format")] = "json",
+) -> TaskOut | str | Any:
+    # Parameter validations
+    if (test_url is None and test_name is not None) or (test_url is not None and test_name is None):
+        return "Invalid arguments!"
+    if (plan_url is None and plan_name is not None) or (plan_url is not None and plan_name is None):
+        return "Invalid arguments!"
+    if plan_url is None and plan_name is None and test_url is None and test_name is None:
+        return "Missing arguments!"
+        # TODO: forward to docs
+    # Disable Celery if not needed
+    if os.environ.get("USE_CELERY") == "false":
+        return service.main(
+            test_url=test_url,
+            test_name=test_name,
+            test_ref=test_ref,
+            plan_url=plan_url,
+            plan_name=plan_name,
+            plan_ref=plan_ref,
+            out_format=out_format,
+            test_path=test_path,
+            plan_path=plan_path)
+    r = service.main.delay(
+            test_url=test_url,
+            test_name=test_name,
+            test_ref=test_ref,
+            plan_url=plan_url,
+            plan_name=plan_name,
+            plan_ref=plan_ref,
+            out_format=out_format,
+            test_path=test_path,
+            plan_path=plan_path)
+    # Special handling of response if the format is html
+    if out_format == "html":
+        status_callback_url = f'{os.getenv("API_HOSTNAME")}/status/html?task-id={r.task_id}'
+        return HTMLResponse(content=html_generator.generate_status_callback(r, status_callback_url))
+    else:
+        return _to_task_out(r)
+
+
+@app.get("/status")
+def status(task_id: Annotated[str | None,
+            Query(
+                alias="task-id",
+                title="Task ID",
+            )
+        ]) -> TaskOut | str:
+    r = service.main.app.AsyncResult(task_id)
+    return _to_task_out(r)
+
+@app.get("/status/html")
+def status_html(task_id: Annotated[str | None,
+            Query(
+                alias="task-id",
+                title="Task ID",
+            )
+        ]) -> HTMLResponse:
+    r = service.main.app.AsyncResult(task_id)
+    status_callback_url = f'{os.getenv("API_HOSTNAME")}/status/html?task-id={r.task_id}'
+    return HTMLResponse(content=html_generator.generate_status_callback(r, status_callback_url))
+
+
+def _to_task_out(r: AsyncResult) -> TaskOut:  # type: ignore [type-arg]
+    return TaskOut(
+        id=r.task_id,
+        status=r.status,
+        result=r.traceback if r.failed() else r.result,
+        status_callback_url=f'{os.getenv("API_HOSTNAME")}/status?task-id={r.task_id}'
+    )
+
+@app.get("/health")
+def health_check():
+    return {"status": "healthy"}
```

Comment:
'healthy' is not exactly correct, it should probably be "up" or "online", but hopefully we'll implement some proper health checks in future PRs :)